### PR TITLE
Clippy-up NetBricks ~ https://github.com/rust-lang/rust-clippy

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
* few instances where we turn off clippy on certain modules, mostly around unsafe use cases for us
* mostly trivial changes
* cleanup of build.sh while accounting for adding lint/clippy
* integrate into travis
* uses 2/3 cargo clippy checks, as the other was impossible to handle (deps containing diff versions of inner libs)
* fix a comment w/ the new pseudo hdr work
* includes some module re-arranging too (as per clippy)

### CODE REVIEW

File: examples/acl-fw/Cargo.toml [hunk 1]
```diff
@@ -2,10 +2,16 @@
...
+repository = "https://github.com/williamofockham/NetBricks/tree/master/examples/acl-fw"
+readme = "..."
+keywords = ["netbricks", "network-functions", "nfs", "packet-processing"]
+categories = ["network-functions", "framework"]
 
 [dependencies]
-fnv = "*"
-lazy_static = "*"
+fnv = ">= 1.0.6"
+lazy_static = ">= 1.3.0"
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply examples/acl-fw/Cargo.toml]: They rec no wildcards actually. https://rust-lang.github.io/rust-clippy/master/#wildcard_dependencies. I can allow this tho, lmk

File: examples/nat-tcp-v4/src/main.rs [hunk 2]
```diff
@@ -126,7 +125,7 @@ fn nat(packet: RawPacket, nat_ip: Ipv4Addr) -> Result<Tcp<Ipv4>> {
                 flow_vec[assigned_port as usize].flow = flow;
                 flow_vec[assigned_port as usize].used = true;
 
-                let mut outgoing_flow = flow.clone();
+                let mut outgoing_flow = flow;
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply examples/nat-tcp-v4/src/main.rs]: Yep. That lint rule, says it’s unnecessary.

File: framework/src/config/mod.rs [hunk 3]
```diff
@@ -151,25 +151,34 @@ impl Source for CommandLine {
         if CLI_ARGS.is_present("primary_core") {
             let core = value_t!(CLI_ARGS, "primary_core", i32)
                 .map_err(|err| ConfigError::Foreign(Box::new(err)))?;
-            map.insert("primary_core".to_string(), Value::new(uri, core as i64));
+            map.insert("primary_core".to_string(), Value::new(uri, i64::from(core)));
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply framework/src/config/mod.rs]: I can turn it off, but it’s one that’s not even pedantic.

File: framework/src/native/mod.rs [hunk 4]
```diff
@@ -1,2 +1,3 @@
-pub(crate) mod libnuma;
-pub(crate) mod zcsi;
+pub mod libnuma;
+pub mod mbuf;
+pub mod zcsi;
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply framework/src/native/mod.rs]: Related to https://rust-lang.github.io/rust-clippy/master/#module_inception. Panda had this avoiding clippy here, but lmk what you think

File: framework/src/packets/icmp/v6/ndp/options/mod.rs [hunk 5]
```diff
@@ -1,13 +1,16 @@
 use fallible_iterator::FallibleIterator;
-use native::zcsi::MBuf;
+use native::mbuf::MBuf;
 use packets::{buffer, ParseError};
 
 pub use self::link_layer_addr::*;
 pub use self::mtu::*;
 pub use self::prefix_info::*;
 
+#[allow(clippy::not_unsafe_ptr_arg_deref, clippy::mut_from_ref)]
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply framework/src/packets/icmp/v6/ndp/options/mod.rs]: By this you mean atop the module instead yeah?

File: examples/acl-fw/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply examples/acl-fw/Cargo.toml]: yeah `*` is not good, what I mean is `lazy_static = ">= 1.3"`, drop the last revision number.

File: examples/nat-tcp-v4/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply examples/nat-tcp-v4/src/main.rs]: but what does this do? does it implicitly clone flow or does it just make flow mutable?

File: framework/src/config/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply framework/src/config/mod.rs]: I'm fine either way. I see a lot of rust code use `as` pretty liberally for casting integer primitives.

File: framework/src/native/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply framework/src/native/mod.rs]: ok the rule is fine. what I mean is you dropped `pub(crate)`, these mods are scoped to be internal to the crate.

File: framework/src/packets/icmp/v6/ndp/options/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply framework/src/packets/icmp/v6/ndp/options/mod.rs]: yeah with the bang `#![allow(...)]`

File: examples/acl-fw/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply examples/acl-fw/Cargo.toml]: actually while on this topic, maybe we should just fix it to a `major.minor` version, cuz thinking about things like `crossbeam` that totally breaks compatibility on version changes.

File: framework/src/config/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply framework/src/config/mod.rs]: I'll go w/ clippy's perceived notion of it for now.

File: framework/src/packets/raw.rs [hunk 6]
```diff
@@ -1,7 +1,8 @@
 use common::Result;
-use native::zcsi::{mbuf_alloc, MBuf};
-use packets::{buffer, Header, Packet};
+use native::mbuf::MBuf;
+use native::zcsi::mbuf_alloc;
 use packets::buffer::read_slice;
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply framework/src/packets/raw.rs]: i like this unsafe ones using `::`

File: examples/acl-fw/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply examples/acl-fw/Cargo.toml]: I think we can kinda do on a library basis right now, esp if some bite us. for now. as we get closer to real prod, then we can be more careful then.

File: examples/nat-tcp-v4/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply examples/nat-tcp-v4/src/main.rs]: implicit.

File: framework/src/utils/mod.rs [hunk 7]
```diff
@@ -73,10 +73,10 @@ mod tests {
     // test flipping a bit at a known position
     #[test]
     fn flippin_bits() {
-        let original: u8 = 0b01101000;
+        let original: u8 = 0b0110_1000;
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply framework/src/utils/mod.rs]: @drunkirishcoder we use them in og for setting/getting bits on segments. So, bringing them back as they are general enough.

[the remark this answers is not in the document]
Comment (drunkirishcoder) [reply framework/src/utils/mod.rs]: maybe we should consider bitmasking there too? I don't know how rustc optimizes these. but bitmasking with constants probably takes less instructions than shifting then masking. But this is ok for now.

**Pull request merged**